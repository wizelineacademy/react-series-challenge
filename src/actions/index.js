import homeActions from './home_actions';
import favoriteActions from './favorites_actions';
import detailsActions from './details_actions';
import listActions from './list_actions';
import searchActions from './search_actions';
import errorActions from './error_actions';
import loadingActions from './loading_actions';
import paginatorActions from './paginator_actions';

export default {
  ...homeActions,
  ...favoriteActions,
  ...detailsActions,
  ...listActions,
  ...searchActions,
  ...errorActions,
  ...loadingActions,
  ...paginatorActions
}
