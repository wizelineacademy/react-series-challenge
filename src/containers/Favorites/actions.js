import { createAction } from 'redux-actions';

const FAVORITE_ID_ADD = 'FAVORITE_ID_ADD';
const FAVORITE_ID_REMOVE = 'FAVORITE_ID_REMOVE';
const FAVORITES_LOAD = 'FAVORITES_LOAD';
const FAVORITES_LOADING_SET = 'FAVORITES_LOADING_SET';
const FAVORITES_SET = 'FAVORITES_SET';
const FAVORITES_SEARCH = 'FAVORITES_SEARCH';

const addFavoriteId = createAction(FAVORITE_ID_ADD);
const removeFavoriteId = createAction(FAVORITE_ID_REMOVE);
const loadFavorites = createAction(FAVORITES_LOAD);
const setLoadingFavorites = createAction(FAVORITES_LOADING_SET);
const setFavorites = createAction(FAVORITES_SET);
const searchFavorites = createAction(FAVORITES_SEARCH);

export default {
  types: {
    FAVORITE_ID_ADD,
    FAVORITE_ID_REMOVE,
    FAVORITES_LOAD,
    FAVORITES_LOADING_SET,
    FAVORITES_SET,
    FAVORITES_SEARCH,
  },
  creators: {
    addFavoriteId,
    removeFavoriteId,
    loadFavorites,
    setLoadingFavorites,
    setFavorites,
    searchFavorites,
  },
};
