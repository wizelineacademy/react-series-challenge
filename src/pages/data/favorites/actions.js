export const TOGGLE_FAVORITE = "TOGGLE_FAVORITE";

export const toggleFavorite = element => (
  { type: TOGGLE_FAVORITE, payload: { element }}
);
