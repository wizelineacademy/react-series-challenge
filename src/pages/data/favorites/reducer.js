import {
  TOGGLE_FAVORITE,
  FILTER_FAVORITE,
} from './actions';

const initialState = {
  list: {},
  filteredElements: [],
  filtered: false,
};

const reducer = (state = initialState, action) => {
  switch (action.type) {
    case TOGGLE_FAVORITE:
      const element = action.payload.element;
      const newFavorites = state.list;
      if(newFavorites[element.id]){
        delete newFavorites[element.id];
      } else {
        newFavorites[element.id]={...element};
      }
      return {
        ...state,
        list: newFavorites,
      };
    case FILTER_FAVORITE: {
      const text = action.payload.text;
      const myRe = new RegExp(text, 'g');
      const newList = Object.values(state.list).filter(value => {
        return value.title.match(myRe);
      })
      return {
        ...state,
        filteredElements: newList,
        filtered: text !== '',
      };
    }
    default:
      return state;
  }
};

export default reducer;
