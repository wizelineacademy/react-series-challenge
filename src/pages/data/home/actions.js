import { asyncActionType } from '../utils';

export const FETCH_TRENDING_GIFS = asyncActionType("FETCH_TRENDING_GIFS");

export const fetchTrendingGifs = () => (
  { type: FETCH_TRENDING_GIFS.PENDING }
);
export const fetchTrendingGifsSuccess = () => (
  { type: FETCH_TRENDING_GIFS.SUCCESS }
);
export const fetchTrendingGifsError = () => (
  { type: FETCH_TRENDING_GIFS.ERROR }
);
