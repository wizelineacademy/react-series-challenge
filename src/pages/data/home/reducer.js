import {
  FETCH_TRENDING_GIFS,
  SEARCH_GIFS,
} from './actions';

const initialState = {
  searching: false,
  error: false,
};

const reducer = (state = initialState, action) => {
  switch (action.type) {
    case FETCH_TRENDING_GIFS.PENDING:
    case SEARCH_GIFS.PENDING:
      return {
        ...state,
        searching: true,
        error: false,
      };
    case FETCH_TRENDING_GIFS.SUCCESS:
    case SEARCH_GIFS.SUCCESS:
      return {
        ...state,
        gifs: {
          ...action.data.gifs,
        },
        searching: false,
        error: false,
      };
    case FETCH_TRENDING_GIFS.ERROR:
    case SEARCH_GIFS.ERROR:
      return {
        ...state,
        searching: false,
        error: true,
      };
    default:
      return state;
  }
};

export default reducer;
