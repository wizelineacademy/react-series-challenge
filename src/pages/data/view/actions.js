import { asyncActionType } from '../utils';

export const SET_CURRENT_GIF = "SET_CURRENT_GIF";
export const setCurrentGif = element => (
  { type: SET_CURRENT_GIF, payload: { element } }
);

export const GET_GIF = asyncActionType("GET_GIF");
export const getGifId = id => (
  { type: GET_GIF.PENDING, payload: { id } }
);
export const getGifIdSuccess = (gif) => (
  { type: GET_GIF.SUCCESS, data: { gif }}
);
export const getGifIdError = (message) => (
  { type: GET_GIF.ERROR, data: { message } }
);
