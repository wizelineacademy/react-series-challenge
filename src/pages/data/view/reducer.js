import {
  SET_CURRENT_GIF,
} from './actions';

const initialState = {};

const reducer = (state = initialState, action) => {
  switch (action.type) {
    case SET_CURRENT_GIF:
      return {
        ...state,
        searching: true,
      };
    default:
      return state;
  }
};

export default reducer;
