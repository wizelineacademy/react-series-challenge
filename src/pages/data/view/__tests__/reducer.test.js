import reducer from '../reducer';
import * as actions from '../actions';

describe('View - Reducer', () => {
  it('Must set a selected gif as current', () => {
    const initialState = reducer(undefined, {});

    const gif = { id: 'a'};

    expect(reducer(
      initialState,
      actions.setCurrentGif(gif),
    )).toMatchSnapshot();
  });

  it('Get Gif By Id Success', () => {
    const initialState = reducer(undefined, {});

    const gif = { id: 'a'};

    expect(reducer(
      initialState,
      actions.getGifIdSuccess(gif),
    )).toMatchSnapshot();
  });

  it('Get Gif By Id Error', () => {
    const initialState = reducer(undefined, {});

    const error = 'Error';

    expect(reducer(
      initialState,
      actions.getGifIdError(error),
    )).toMatchSnapshot();
  });
});
