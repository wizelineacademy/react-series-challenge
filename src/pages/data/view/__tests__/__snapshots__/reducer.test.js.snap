// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`View - Reducer Get Gif By Id Error 1`] = `
Object {
  "currentGif": Object {},
  "error": true,
  "searched": true,
}
`;

exports[`View - Reducer Get Gif By Id Success 1`] = `
Object {
  "currentGif": undefined,
  "error": false,
  "searched": true,
}
`;

exports[`View - Reducer Must set a selected gif as current 1`] = `
Object {
  "currentGif": Object {
    "id": "a",
  },
  "error": false,
  "searched": false,
}
`;
